windowing_library.py

```diff
@@ -0,0 +1,71 @@
+import pickle
+import matplotlib.pyplot as plt
+import numpy as np
+import skimage 
+from scipy import signal
+from scipy.spatial import procrustes
+from skimage.util.shape import view_as_windows
+import statistics
+
+# make it usable for all rows and make one function you call first for phase 1 and 2 then a second main function for phase 3
+
+def create_windows(matrix, rows, window_size, stride, mod_num):
+  x = matrix['x']
+  y = matrix['y']
+  z = matrix['z']
+  nanos = matrix['nanos'] 
+  np_data = np.array([x,y,z,nanos], 'd')
+
+  windowed_data = view_as_windows(np_data, (rows, window_size), step=stride)
+  return windowed_data
+
+def find_initial_lift_times(lift_windows, rows):
```

Comment:
Please add comments to all functions explaining what they are used for and preferably what the input to the function is. If you like, you can also add types to the input.